### PR TITLE
Set protocol component prices

### PR DESCRIPTION
**Pull request by jackiep00**
I've checked that:

* [X] the query produces the intended results
* [X] the folder name matches the schema name
* [X] the schema name exists in Dune
* [X] views are prefixed with `view_`, functions with `fn_`.
* [X] the filename matches the defined view, table or function and ends with .sql
* [X] each file has only one view, table or function defined  
* [X] column names are `lowercase_snake_cased`

This PR is for a daily price feed for the underlying components of Sets created using Set Protocol. These prices are important for estimating TVL and whatnot, and is intended to work across the entire ecosystem.

I've validated the insert function and backfilling function in this query here:
https://dune.xyz/queries/594861
This doesn't run in one go because the backfilling operation takes longer than 30 minutes, but you can execute each insert statement individually.

Most of the operative logic is clarified in the README. Architecturally it's modeled off of insert_prices_from_dex_data, and uses the dex pricing as a key input.

Update: Addressed feedback on putting the token mapping into the PR instead of relying on `dune_user_generated`

### CODE REVIEW

**Comment by jackiep00** [comment]
@jeff-dude Saw your feedback and resubmitted here!

File: ethereum/setprotocol_v2/set_component_token_mapping.sql [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+create table if not exists setprotocol_v2.set_component_token_mappings
+(
+  symbol varchar,
+  component_address bytea,
+  token_type varchar,
+  mapped_symbol varchar,
+  mapped_component_address bytea         
+);
+
+create index if not exists on setprotocol_v2.set_component_token_mappings(component_address, mapped_component_address);
```
Comment (jeff-dude) [inline ethereum/setprotocol_v2/set_component_token_mapping.sql:15]: @jackiep00 i've created this table & inserted the rows, however, the indices were not created due to syntax error. refer to your other table script, where there is a name to be applied before the 'on' command. please fix that syntax and push a commit to this PR, then i'll apply them. i'll continue on without the index here

**Comment by jeff-dude** [comment]
@jackiep00 here is where we stand:

- [ ] set_component_token_mappings table created and populated
- [ ] set_component_token_mappings indices created
- [ ] daily_component_prices table & indices created
- [ ] daily_component_prices history loaded
- [ ] insert function created
- [ ] cron job inserted and active

once complete, we can merge & close

**Comment by jackiep00** [comment]
@jeff-dude alright, I added the name and fixed the syntax error. Sorry about that!

**Review by jeff-dude — APPROVED** [review]

**Pull request merged**